src/graphql/execution/execute.py

```diff
@@ -459,12 +468,16 @@ async def get_results() -> dict[str, Any]:
                 field = awaitable_fields[0]
                 results[field] = await results[field]
             else:
-                results.update(
-                    zip(
-                        awaitable_fields,
-                        await gather(*(results[field] for field in awaitable_fields)),
-                    )
-                )
+                tasks = {}
+                for field in awaitable_fields:
+                    tasks[create_task(results[field])] = field  # type: ignore[arg-type]
+
+                done, pending = await wait(tasks, return_when=FIRST_EXCEPTION)
+                for task in pending:
+                    task.cancel()
+
+                results.update((tasks[task], task.result()) for task in done)
```

Comment:
This maintains the previous behaviour of not setting any results if an exception is raised, but I could change this to set the results we do have and then raise the exception.

Reply:
Does this behave in-line with the spec regarding partial results?

https://spec.graphql.org/draft/#sec-Handling-Execution-Errors

Reply:
I think this is fine because the potentially completed fields are siblings of the field with an exception, and the exception is propagated to the parent which will either be `null` or propagate further up. In either case these siblings field won't be present in the response.